### ISSUE TITLE
[16.0] payroll_account support in hr_payslip.py for multicurrency

### ISSUE THREAD

**Issue by pleirb** [OCA/payroll#167]
Hi everybody how are you?

What do you think or what is suggested to be done if the payroll has to be generated not in base currency?
Currently in the action_payslip_done method the value of the currency is obtained in this way:
```
currency = (
  slip.company_id.currency_id or slip.journal_id.company_id.currency_id
)
```
Reference: https://github.com/OCA/payroll/blob/d0196f8cda6a0938cb742bcdb63f322857f6a896/payroll_account/models/hr_payslip.py#L61
It wouldn't be more real if you obtained the currency in this way?:
```
currency = (
  slip.currency_id or slip.journal_id.currency_id
)
```
Of course, for this it would be necessary to have the following module installed, which is still under review in version 17.0: [**hr_contract_currency**](https://github.com/OCA/hr/pull/1357)
On the other hand, the values ​​of currency_id and amount_currency would have to be loaded into the debit and credit lines and then perform the corresponding reconciliation.
I am not an expert programmer or accountant but I have carried out some tests in a local development environment (in version 16.0) also using two old modules in version 8.0 of Odoo developed by ClearCorp that I migrated to version 16.0 and thus be able to contemplate this multi-currency scenario:
[**hr_payroll_multicurrency**](https://github.com/Vauxoo/addons-clearcorp/tree/8.0/hr_payroll_multicurrency)
This module was written to extend the functionality of payroll to support multi-currency
[**hr_payroll_account_multicurrency**](https://github.com/Vauxoo/addons-clearcorp/tree/8.0/hr_payroll_account_multicurrency)
This module was written to extend the functionality of payroll to create journal items in multi-currency

Basically I would like to know from the main people in charge of this repository if there are intentions to incorporate multi-currency support for payrolls and if not, receive an option on the path I am following to know if it is worth contributing to what I am doing. and that others can collaborate so that this functionality is present in version 16 and later

From already thank you very much

**Comment by nimarosa** [comment]
@pleirb Hello, i just see this issue, didn't see it before. 
It's a interesting change what you are proposing, we have to see if it will have any impact if you have payslips in multiple currencies because sometimes payslips use information from another payslips. 

I tend to be more aligned to have the payslips in company currency since it's not a common case to have payslips in more than one currency. But if you want to submit a PR proposing your change we can review it in detail.

**Comment by pleirb** [comment]
> [Hola](https://github.com/pleirb), acabo de ver este problema, no lo vi antes. Es un cambio interesante lo que usted está proponiendo, tenemos que ver si tendrá algún impacto si usted tiene recibo de pago en múltiples monedas porque a veces los recibos de pago utilizan información de otra nómina.
> 
> Tiendo a estar más alineado para tener los recibos de pago en la moneda de la compañía ya que no es un caso común tener recibos de pago en más de una moneda. Pero si quieres enviar una PR proponiendo tu cambio podemos revisarlo en detalle.

At some point I will be able to concentrate on this, surely at the beginning of winter (perhaps from August) I am currently very busy. I think there are advances regarding the multi-currency landscape that could be taken advantage of. In Uruguay, for example, it is very common to have the dollar as the base currency for both purchases and sales, but the payment of salaries must be in national currency (pesos). I suppose this scenario is recurrent in other countries.
